### PR TITLE
Feat(medusa): cancel order edit

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- cancel an order edit

Fixes CORE-500

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/order-edits/cancel-order-edit.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import { Request, Response } from "express"
+import { OrderEditService } from "../../../../services"
+import { IsOptional, IsString } from "class-validator"
+import { EntityManager } from "typeorm"
+
+/**
+ * @oas [post] /order-edits
+ * operationId: "PostOrderEdits"
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/order-edits/cancel-order-edit.ts:8]:
```suggestion
 * @oas [post] /order-edits/{id}/cancel
 * operationId: "PostOrderEditsOrderEditCancel"
```

Comment (srindom) [reply packages/medusa/src/api/routes/admin/order-edits/cancel-order-edit.ts:8]: Should also include `id` as a param

**Review by fPolic — APPROVED** [review]
LGTM 💪

File: packages/medusa/src/services/order-edit.ts [hunk 2]
```diff
@@ -382,4 +383,40 @@ export default class OrderEditService extends TransactionBaseService {
 
     return orderEdit
   }
+
+  async cancel(orderEditId: string, userId: string): Promise<OrderEdit> {
```
Comment (srindom) [inline packages/medusa/src/services/order-edit.ts:387]: todo: (sorry for not spotting in the first iteration) It should technically be fine to cancel an order without it being a user who does it (e.g., in a subscriber)

```suggestion
  async cancel(orderEditId: string, userId?: string): Promise<OrderEdit> {
```

File: packages/medusa/src/api/routes/admin/order-edits/cancel-order-edit.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+ * x-codeSamples:
+ *   - lang: JavaScript
+ *     label: JS Client
+ *     source: |
+ *       import Medusa from "@medusajs/medusa-js"
+ *       const medusa = new Medusa({ baseUrl: MEDUSA_BACKEND_URL, maxRetries: 3 })
+ *       // must be previously logged in or use api token
+ *       medusa.admin.orderEdit.cancel(orderEditId)
+ *         .then(({ order_edit }) => {
+ *           console.log(order_edit.id);
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/order-edits/cancel-order-edit.ts:23]: nit(non-blocking):
```suggestion
 *           console.log(order_edit.id)
```

File: packages/medusa/src/api/routes/admin/order-edits/cancel-order-edit.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
+import { Request, Response } from "express"
+import { OrderEditService } from "../../../../services"
+import { IsOptional, IsString } from "class-validator"
+import { EntityManager } from "typeorm"
+
+/**
+ * @oas [post] /order-edits
+ * operationId: "PostOrderEdits"
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/order-edits/cancel-order-edit.ts:8]:
```suggestion
 * @oas [post] /order-edits/{id}/cancel
 * operationId: "PostOrderEditsOrderEditCancel"
```

File: packages/medusa/src/api/routes/admin/order-edits/cancel-order-edit.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+export default async (req: Request, res: Response) => {
+  const { id } = req.params
+
+  const orderEditService = req.scope.resolve(
+    "orderEditService"
+  ) as OrderEditService
+
+  const manager = req.scope.resolve("manager") as EntityManager
+
+  const userId = req.user?.id ?? req.user?.userId ?? ""
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/order-edits/cancel-order-edit.ts:62]: nit(non-blocking): you can't reach the handler without being authenticated
```suggestion
  const userId = req.user.id ?? req.user.userId
```

File: packages/medusa/src/services/order-edit.ts (hunk 2 above)
Comment (srindom) [reply packages/medusa/src/services/order-edit.ts:387]: also maybe we should align on either `userId` like here or `{ loggedInUser }` as seen elsewhere; have no opinion

**Review by srindom — APPROVED** [review]
LGTM! 🎉

**Pull request merged**